mirdata/datasets/filosax.py

```diff
@@ -0,0 +1,512 @@
+        )
+        self.pitch_average_val = (
+            input_dict["pitch_average_val"]
+            if "pitch_average_val" in input_dict
+            else 0.0
+        )
+        self.pitch_average_time = (
+            input_dict["pitch_average_time"]
+            if "pitch_average_time" in input_dict
+            else 0.0
+        )
+        self.pitch_curve = (
+            input_dict["pitch_curve"] if "pitch_curve" in input_dict else [0.0]
+        )
+        self.pitch_vib_freq = (
+            input_dict["pitch_vib_freq"] if "pitch_vib_freq" in input_dict else 0.0
+        )
+        self.pitch_vib_ext = (
+            input_dict["pitch_vib_ext"] if "pitch_vib_ext" in input_dict else 0.0
+        )
+        self.spec_cent = input_dict["spec_cent"] if "spec_cent" in input_dict else 0.0
+        self.spec_flux = input_dict["spec_flux"] if "spec_flux" in input_dict else 0.0
+        self.spec_cent_curve = (
+            input_dict["spec_cent_curve"] if "spec_cent_curve" in input_dict else [0.0]
+        )
+        self.spec_flux_curve = (
+            input_dict["spec_flux_curve"] if "spec_flux_curve" in input_dict else [0.0]
+        )
+        self.seq_len = input_dict["seq_len"] if "seq_len" in input_dict else -1
+        self.seq_num = input_dict["seq_num"] if "seq_len" in input_dict else -1
+
+
+class Track(core.Track):
+    """Filosax track class
+    # -- YOU CAN AUTOMATICALLY GENERATE THIS DOCSTRING BY CALLING THE SCRIPT:
+    # -- `scripts/print_track_docstring.py my_dataset`
+    # -- note that you'll first need to have a test track (see "Adding tests to your dataset" below)
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        audio_path (str): path to audio file
+        annotation_path (str): path to annotation file
+        midi_path (str): path to MIDI file
+        musicXML_path (str): path to musicXML file
+        pdf_path (str): path to PDF file
+
+    Cached Properties:
+        notes ([Note]): an ordered list of Note objects
+
+    """
+
+    def __init__(self, track_id, data_home, dataset_name, index, metadata):
+
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.annotation_path = self.get_path("annotation")
+        self.midi_path = self.get_path("midi")
+        self.musicXML_path = self.get_path("musicXML")
+        self.pdf_path = self.get_path("pdf")
+
+    @core.cached_property
+    def notes(self) -> Optional[List[Note]]:
+        """The track's note list - only for Sax files
+
+        Returns:
+            * [Note] - ordered list of Note objects
+
+        """
+        note_path = "" if self.annotation_path == None else self.annotation_path
+        if note_path == "":
+            print("Note data only available for Sax tracks.")
+            return [Note({})]
```

Comment:
This is slightly convoluted, due to attempts to fix the Exception error (which you found to be somewhat unrelated to this). So now, if the user tries to get note annotations from the backing track, it fails extremely elegantly - with an empty list of `Note` objects. Is this okay?

Reply:
Yes, sure! I think I remember now. Then, would it be possible to add a small test to make sure that `[Note({})]` is returned when the path is None?

Another thing I'd like to mention is that (and I'm sorry to detect that now), I think that it'd be good to use `load_annotation()` here, and mention in the docstring of `load_annotation()` that it can also be used to load the notes which are only available for the sax track. 
Then `notes()` would look like this:

```
@core.cached_property
def notes(self) -> Optional[List[Note]]:
    """ The track's note list - only for Sax files
    Returns:
        * [Note] - ordered list of Note objects
    """
    if not self.annotation_path:
        print("Note data only available for Sax tracks.")
        return [Note({})]
    else:        
        return load_annotation(self.load_annotation)
```

Does that sound reasonable to you? I understand why to use note_path but I think it is a little bit overkill and you could briefly explain that in the docstring without the need of duplicating code :) Let me know what you think!

At some point, I also think that the print here may not be needed, as long as the warning (`"Note data only available for Sax tracks."`) is clearly stated in the docstring. Maybe it'll be a bit annoying to get this message multiple times once you have learned that only sax tracks have note data.

Well, I'm sorry to leave here that piece of text @dave-foster... I hope that this helps and I'm open to discussing these last details before the loader is merged. Thanks for addressing the rest of my review, I went through the docs, and looks ready!!

Reply:
I've implemented your version of the code (which is far better than what was there), and added a test